components/scream/src/share/io/scorpio_output.hpp

```diff
@@ -0,0 +1,637 @@
+ *
+ *  At construction time ALL output instances require
+ *  1. an EKAT comm group and
+ *  2. a EKAT parameter list
+ *  3. a shared pointer to the field repository
+ *  4. a shared pointer to the grids manager
+ * 
+ *  The EKAT parameter list must contain all of the top level output control information and follows the format:
+ *  ------
+ *  FILENAME: STRING
+ *  AVERAGING TYPE: STRING
+ *  GRID: STRING                   (optional)
+ *  FREQUENCY:
+ *    OUT_N: INT
+ *    OUT_OPTION: STRING
+ *    OUT_MAX_STEPS: INT
+ *  FIELDS:
+ *    Number of Fields: INT
+ *    field 1: STRING
+ *    ...
+ *    field N: STRING
+ *  restart_hist_N: INT            (optional)
+ *  restart_hist_OPTION: STRING    (optional)
+ *  RESTART FILE: BOOL             (optional)
+ *  -----
+ *  where,
+ *  FILENAME is a string of the filename suffix.  TODO: change this to a casename associated with the whole run.
+ *  AVERAGING TYPE is a string that describes which type of output, current options are:
+ *    instant - no averaging, output each snap as is.
+ *    average - average of the field over some interval described in frequency section.
+ *    min     - minimum value of the field over some time interval
+ *    max     - maximum value of the field over some time interval
+ *  GRID is a string describing which grid to write on, currently the only option is the default Physics.
+ *  FREQUENCY is a subsection that controls the frequency parameters of output.
+ *    OUT_N is an integer of the frequency of output given the units from OUT_OPTION
+ *    OUT_OPTION is a string for the units of output frequency, examples would be "Steps", "Months", "Years", "Hours", etc.
+ *    OUT_MAX_STEPS is an integer of the maximum number of steps that can exist on a single file (controls files getting too big).
+ *  FIELDS is a subsection that lists all the fields in this output stream.
+ *    Number of Fields is an integer that specifies the number of fields in this output stream.
+ *    field 1,...,field N is a list of each field in the output stream by name in field manager.
+ *  restart_hist_N is an optional integer parameter that specifies the frequenct of restart history writes.
+ *  restart_hist_OPTION is an optional string parameter for the units of restart history output.
+ *  RESTART FILE is an optional boolean parameter that specifies if this output stream is a restart output, which is treated differently.
+ *
+ *  Usage of this class is to create an output file, write data to the file and close the file.
+ *  This class keeps a running copy of data for all output fields locally to be used for the different averaging flags.
+ * --------------------------------------------------------------------------------
+ *  (2020-10-21) Aaron S. Donahue (LLNL)
+ */
+
+/* There are a number of things that need to be cleaned up - in no particular order -
+ *--1) Gather dimension data from grid manager or field repo.  We may need to use field repo since grid manager
```

Comment:
Are the items in this list reflected in GitHub issues? That's a better place for a TODO. At the very least, we can erase the entries here that are --DONE--